### ISSUE TITLE
Can't add Google Account to MicroG

### ISSUE THREAD

**Issue by alexjmccoy** [microg/android_packages_apps_GmsCore#1082]
On a brand new install of CalyxOS with microG I cannot add a Google Account. I am running version `0.2.10.19420-2-vtm` of microG Services Core. I am using TOTP with my Google account.

When I select Add Google Account I am provided with the Google sign-in page and I can successfully enter my Google account address, password, and TOTP code. On the next screen I Agree to Terms of Service, and then finally I get a message that says "Sorry... There was a problem communicated with Google servers. Try again later.

I am able to sign into the browser on the devices so it seems to be a microG issue.

![485-1024](https://user-images.githubusercontent.com/15354332/86344533-60627380-bc28-11ea-8648-ee6a4d022563.png)

**Comment by alexjmccoy** [comment]
Just an update: I have tried to add accounts via both:

* microG > Account (Add Google account) and
* Settings > Accounts > + Add account > Google

the latter of which launches the same API. No dice either way. I have two accounts: my main Google account and also a test/burner account with no 2FA. Neither accounts works using either method.

I have also tried clearing microG storage, factory resets, using a VPN (obviously also NOT using a VPN) etc. Same error message every time.

**Comment by alexjmccoy** [comment]
My Google account does register that I am trying to sign in. I went into my Google Account > Security > Manage my devices and signed out of every device. Went into microG, tried to add my account, and went through the prompts. I received the same error: "There was a problem communicating with Google servers. Try again later."

After this, I went to my Google account and looked at the devices that are currently signed in:

![Screenshot 2020-07-13 at 9 38 48 AM](https://user-images.githubusercontent.com/15354332/87311714-9b4c8d00-c4ed-11ea-8544-bc50f6c19059.png)

**Comment by alexjmccoy** [comment]
I managed to get this working with the help of one of the developers of CalyxOS. Here is the only successful algorithm that has worked every time it was tried:

1. Factory reset
2. Setup device, don't connect to wifi or mobile data / don't put a SIM in.
3. microG elected, no other apps selected
4. go to microG Settings, disable device registration, cloud messaging and safetynet
5. Enable and setup wifi
6. Go to Settings -> Accounts, add account
7. Works, never worked on this device before

I have rebooted the device, re-enabled all the microG toggles and it still works. Everything seems to be good now.
